public/app/features/alerting/AlertTabCtrl.ts

```diff
@@ -318,6 +334,7 @@ export class AlertTabCtrl {
         return this.$q.when(result);
       }
     }
+    return undefined;
```

Comment:
why did you add these?

Reply:
i add type for $q: IQService
and here will get ts error: for not all branch return value